### DISCUSSION TITLE
How to change tempo and how to loop a section.

### DISCUSSION THREAD

**Discussion by eyeke04** [cifkao/html-midi-player#47] [Q&A] [answered]
Hi Cifka,

Thank you again for this amazing library. Is there a `player` method to increase/reduce the tempo.

Also how can I loop through a section of the noteSequence?

Extra question - please when the player start, I want to get the first note in the sequence that will be played. For example, if I pause the player and I want to restart it, I want get the very first note that will be played.

**Comment by cifkao** [comment, marked as answer]
Thanks for your question! The player API is described [here](https://github.com/cifkao/html-midi-player/blob/master/doc/midi-player.md#midi-player). Whatever is not there isn't directly supported, but of course some advanced features can be implemented using this API.

> Is there a `player` method to increase/reduce the tempo.

Unfortunately there isn't, see #21.

> Also how can I loop through a section of the noteSequence?

Looping is not supported, see #8. You could try listening for the `note` event and when the time exceeds the end of your section, you reset back to the beginning (by setting `player.currentTime`). However as I noted in #8, the looping probably won't be perfect, there will be some delay.

> Extra question - please when the player start, I want to get the first note in the sequence that will be played. For example, if I pause the player and I want to restart it, I want get the very first note that will be played.

You can loop through `player.noteSequence.notes` and find the first note after `player.currentTime`. Or you can listen to the `note` event and you will be notified when a note plays (you can stop listening after the first one).

BTW, don't hesitate to create multiple discussion threads if you have multiple questions.

**Reply by eyeke04** [reply]
thank you 🙏🏿
